src/libs/actions/Report.js

```diff
@@ -298,19 +298,26 @@ function fetchIOUReportID(debtorEmail) {
 }
 
 /**
- * Fetches chat reports when provided a list of
- * chat report IDs
- *
+ * Fetches chat reports when provided a list of chat report IDs.
+ * If the shouldRedirectIfInacessible flag is set, we redirect to the Concierge chat
+ * when fetching a single chat that is inacessible.
  * @param {Array} chatList
+ * @param {Boolean} shouldRedirectIfInacessible
  * @returns {Promise<Number[]>} only used internally when fetchAllReports() is called
  */
-function fetchChatReportsByIDs(chatList) {
+function fetchChatReportsByIDs(chatList, shouldRedirectIfInacessible = false) {
     let fetchedReports;
     const simplifiedReports = {};
     return API.GetReportSummaryList({reportIDList: chatList.join(',')})
-        .then(({reportSummaryList}) => {
+        .then(({reportSummaryList, jsonCode}) => {
             Log.info('[Report] successfully fetched report data', true);
             fetchedReports = reportSummaryList;
+
+            // If we receive a 404 response while fetching a single report, treat that report as inacessible.
+            if (jsonCode === 404 && chatList.length === 1) {
+                throw new Error('inacessible');
```

Comment:
Can we perhaps a constant for this and maybe give it a slightly more descriptive error something like this?

```js
const CONST = {
    ...,
    ERROR: {
        INACCESSIBLE_REPORT: 'Report not found',
    },
};
```

Reply:
Then we can use `CONST.ERROR.INACCESSIBLE_REPORT` in these two locations.